File: YEAR 2/SEM 1/Advanced Programming Methods/LAB 4/src/Model/Expression/ConstantExpression.java
package Model.Expression;

import Util.Dictionary.MyIDictionary;
import Util.Heap.IHeap;

public class ConstantExpression implements IExpression {
    private int value;

    public ConstantExpression(int value){
        this.value = value;
    }

    @Override
    public int evaluate(MyIDictionary<String,Integer> symbolTable, IHeap heap) throws ExpressionException {
        return this.value;
    }

    @Override
    public String toString() {
        return "(" + this.value + ")";
    }
}

File: YEAR 2/SEM 1/Advanced Programming Methods/ToyLanguage/ToyLanguage/Model/Expression/ConstantExpression.cs
using ToyLanguage.Util.Dictionary;

namespace ToyLanguage.Model.Expression
{
    class ConstantExpression : IExpression
    {
        private readonly int number;

        public ConstantExpression(int number)
        {
            this.number = number;
        }

        public int Evaluate(MyIDictionary<string, int> symbolTable)
        {
            return number;
        }

        public override string ToString()
        {
            return "ConstantExpression(" + number + ")";
        }
    }
}
